test/api/support/ffactory.js

```diff
@@ -0,0 +1,98 @@
+const Promise = require("bluebird")
```

Comment:
I can look for mixins to replace the helpers we use. I'd like to see us using the native promises as well.